lib/whitelist/whitelist.rb

```diff
@@ -0,0 +1,43 @@
+class Whitelist
+
+  def initialize(whitelist)
+    @whitelist = whitelist
+  end
+
+  def apply(check_name, headers, rows)
+    check = get_or_else(@whitelist, check_name, {})
+    rules = get_or_else(check, 'rules', [])
+    rule_predicates = rules.map { |rule| Whitelist.predicate_for(headers, get_or_else(rule, 'predicate', [])) }
+    rows.reject { |row| rule_predicates.any? { |r| r.call(row) } }
+  end
+
+  def get_or_else(hash, key, default)
+    # can't use #fetch since it returns nil if the key exists but maps to nil
+    value = hash[key]
+    value ? value : default
+  end
+
+  def report_expired_entries(reference_date)
+    expiries = @whitelist.flat_map { |check| get_or_else(check[1], 'rules', []).map { |rule| [check[0], rule['expiry'], rule['reason']]} }
```

Comment:
Sure. I guess everyone has different syntax highlighting and tolerance for this sort of nonsense :)